web-portal/backend/src/utils/utils.service.ts

```diff
@@ -168,39 +169,64 @@ export class UtilsService {
 
   @Cron(CronExpression.EVERY_MINUTE)
   async watchEvent() {
-    console.log('Getting Event Logs');
-    const blockNumber = await viemClient.getBlockNumber();
-    const logs = await viemClient.getLogs({
-      event,
-      address: portrAddress,
-      fromBlock: blockNumber - BigInt(1000),
-      toBlock: blockNumber,
-    });
+    console.log('Getting Event Logs from all clients');
+
+    // Define clients and their respective names
+    const clients = [
+      { client: opClient, name: 'optimism' },
+      { client: gnosisClient, name: 'gnosis' },
+      { client: baseClient, name: 'base' }
+    ];
+
+    // Fetch and parse logs for all clients
+    const allParsedLogs = await Promise.all(
+      clients.map(async ({ client, name }) => {
+        console.log(`Getting Event Logs from ${name}`);
+        const blockNumber = await client.getBlockNumber();
+        const logs = await client.getLogs({
+          event,
+          address: portrAddress,
+          fromBlock: blockNumber - BigInt(1000),
+          toBlock: blockNumber,
+        });
+
+        return this.parseLogs(logs, name);
+      })
+    );
 
-    const parsedLogs: IParsedLog[] = logs.map((log: any) => ({
-      tenantId: fromHex(log?.args?._identifier, 'string').replaceAll(
-        `\x00`,
-        '',
-      ),
-      amount: Number(log?.args?._amount),
-      referenceId: log.transactionHash!,
-      transactionType: TransactionType.CREDIT!,
-    }));
+    const [parsedLogsOP, parsedLogsGnosis, parsedLogsBase] = allParsedLogs;
 
-    console.log({ parsedLogs });
+    console.log({ parsedLogsOP, parsedLogsGnosis, parsedLogsBase });
 
-    if (!parsedLogs) console.log('No New Redemptions');
+    if (!parsedLogsOP.length && !parsedLogsGnosis.length && !parsedLogsBase.length) {
+      console.log('No New Redemptions');
+      return;
+    }
 
     // Create records for unique logs
     const appliedLogs = await this.prisma.client.paymentLedger.createMany({
       skipDuplicates: true,
-      data: parsedLogs,
+      data: [...parsedLogsOP, ...parsedLogsGnosis, ...parsedLogsBase],
     });
 
     console.log({ appliedLogs });
 
-    if (!appliedLogs) console.log('Error Applying logs');
+    if (!appliedLogs) {
+      console.log('Error Applying logs');
+    } else {
+      console.log('Applied New logs');
+    }
+  }
 
-    console.log('Applied New logs');
+  // Helper function to parse logs
+  parseLogs(logs: any[], network: string): IParsedLog[] {
+    return logs.map((log: any) => ({
+      tenantId: fromHex(log?.args?._identifier, 'string').replaceAll(`\x00`, ''),
+      amount: Number(log?.args?._amount),
```

Comment:
This needs to be sized correctly. It is currently 18 decimals, will give the user too many credits (currently 1 trillion times too many).